qcodes/dataset/measurements.py

```diff
@@ -272,6 +276,25 @@ def _unbundle_arrayparameter(self,
                                             parameter.setpoints,
                                             res, found_parameters)
 
+    def _unbundle_arrayparameter2(self, parameter: ArrayParameter2,
+                                  res: List[res_type],
+                                  found_parameters: List[str]) -> None:
+        setpoint_names = []
+        setpoint_data = []
+        for setpointparam in parameter.setpoints:
+            these_setpoints = setpointparam.get()
```

Comment:
should we be careful here? getting the values of setpoints implicitly - are there any restrictions on time/performance or on smth else? i'm just worried that if the `.get()` is hidden from the researcher, then something can go wrong, but i do not know what.

Reply:
Yes you are right in the sense that the get method could have all kinds of side effects. For example the get method of the Signal Hound calls a function on the device to get the start,num_points and stepsize

Reply:
indeed. thanks for clearing my feeling :)